### PR TITLE
[14.0][FIX]add filter on pos_reference field when get pos records orders

### PR DESCRIPTION
**Pull request by DorianMAG**
Error get pos orders records.
When pos_reference field is empty regex iteration breaks.
To fix, add filter to exclude empty pos_reference.

### CODE REVIEW

**Comment by flotho** [comment]
Hi @alexis-via we found a strange issue here.
When using your awesome module pos_session_pay_invoice pos order are created with no pos_reference 
then ins the POS when you try to get all the orders with the right button on the right, the method `export_for_ui` crash because it doesn't succeed to apply the regexp here https://github.com/OCA/OCB/blob/14.0/addons/point_of_sale/models/pos_order.py#L710
Totally normal because the pos order you create are not originated from the POS.
So we proposed to not load the pos order coming from your module when you try to get the order history.

**Review by flotho — CHANGES_REQUESTED** [review]
here the pylint will fail
you need a space

File: pos_session_pay_invoice/models/pos_order.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+
+class PosOrder(models.Model):
+    _inherit = "pos.order"
+    
+    @api.model
+    def search_paid_order_ids(self, config_id, domain, limit, offset):
+        domain = [("pos_reference", "!=", False), ("lines", "!=", False)]
+        res = super(PosOrder, self).search_paid_order_ids(config_id, domain, limit, offset)
+        return res
```
Comment (flotho) [inline pos_session_pay_invoice/models/pos_order.py]: same thing here

**Review by flotho — CHANGES_REQUESTED** [review]

**Comment by flotho** [comment]
hi @legalsylvain already faced the same issue ?

File: pos_session_pay_invoice/models/pos_order.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+from odoo import api, models
+
+
+class PosOrder(models.Model):
+    _inherit = "pos.order"
+    
+    @api.model
+    def search_paid_order_ids(self, config_id, domain, limit, offset):
+        domain = [("pos_reference", "!=", False), ("lines", "!=", False)]
```
Comment (legalsylvain) [inline pos_session_pay_invoice/models/pos_order.py]: You should add domain to the current domain. Or did I missed something ?

**Review by legalsylvain — COMMENTED** [review]
I dont use this module for the time being.

File: pos_session_pay_invoice/models/pos_order.py (hunk 2 above)
Comment (flotho) [reply pos_session_pay_invoice/models/pos_order.py]: hum right, we forgot to add instaed of replacing.

**Comment by flotho** [comment]
@DorianMAG thanks for this first step,
Now you should use the AND() operator to combine the orginal domain with the one you want to apply otherwise your cod is breaking the flow of inheritance

File: pos_session_pay_invoice/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from . import pos_order
```
Comment (flotho) [inline pos_session_pay_invoice/models/__init__.py]: missing return here

File: pos_session_pay_invoice/__init__.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
 
 from . import wizard
+from . import models
```
Comment (flotho) [inline pos_session_pay_invoice/__init__.py:4]: missing return here

**Review by flotho — CHANGES_REQUESTED** [review]
@DorianMAG thanks for this first step,
Now you should use the AND() operator to combine the orginal domain with the one you want to apply otherwise your cod is breaking the flow of inheritance

File: pos_session_pay_invoice/models/__init__.py (hunk 3 above)
Comment (DorianMAG) [reply pos_session_pay_invoice/models/__init__.py]: thanks for your comments.
It s done now.
Regards

File: pos_session_pay_invoice/models/pos_order.py (hunk 2 above)
Comment (DorianMAG) [reply pos_session_pay_invoice/models/pos_order.py]: Hi @legalsylvain .
Thank you for your feedback.
It's kinda new to me

**Comment by flotho** [comment]
thanks @DorianMAG

**Review by flotho — APPROVED** [review]

**Comment by DorianMAG** [comment]
FIX already done and merged
https://github.com/OCA/pos/pull/1006

**Comment by DorianMAG** [comment]
FIX already done and merged
https://github.com/OCA/pos/pull/1006

**Pull request closed without merging**